### PR TITLE
refactor csat context processor logic

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: privaterelay/context_processors.py [hunk 1]
```diff
@@ -77,3 +50,43 @@ def _get_fxa(request):
         return fxa
     except AttributeError:
         return None
+
+
+@lru_cache(maxsize=None)
```
Comment (groovecoder) [inline privaterelay/context_processors.py:55]: If nothing else, I'd like to move the logic into a cached function like this so we can avoid the performance hit of all this logic on every request.

File: privaterelay/context_processors.py [hunk 2]
```diff
@@ -77,3 +50,44 @@ def _get_fxa(request):
...
+        days_since = _get_days_since(days_since_args[reason])
+        if reason == 'premium' and days_since == None:
+            return '', None
+        for num_days in day_periods:
+            if (days_since >= num_days):
+                csat_dismissal_cookie = (
+                    f'csat-survey-{reason}-{num_days}days_dismissed'
+                )
+                if (request.COOKIES.get(csat_dismissal_cookie)):
+                    return csat_dismissal_cookie, None
```
Comment (groovecoder) [inline privaterelay/context_processors.py:79]: Dang, this `return` will always execute as soon as someone dismisses the `free7days` or the `premium7days` prompt. We'll never go into the next iteration over the `day_periods` loop. Needs some re-thinking again.

File: privaterelay/context_processors.py [hunk 3]
```diff
@@ -77,3 +50,44 @@ def _get_fxa(request):
...
+                return csat_dismissal_cookie, f'{reason}{num_days}days'
+
+    return '', None
+
+
+def _get_days_since(days_since_arg):
+    now = datetime.now(timezone.utc)
+    if type(days_since_arg == Profile):
```
Comment (Vinnl) [inline privaterelay/context_processors.py:87]: I'm assuming that there's a wrongly placed bracket here?

```suggestion
    if type(days_since_arg) == Profile:
```

File: privaterelay/context_processors.py (hunk 1 above)
Comment (Vinnl) [reply privaterelay/context_processors.py:55]: OK, I did that in 50dbfab5.

**Comment by groovecoder** [comment]
Closing this. Refactor can come later.

**Pull request closed without merging**